main.go

```diff
@@ -381,20 +421,50 @@ func runCLIApp(c *cli.Context) (err error) {
 	{
 		mfs, err = mountWithArgs(bucketName, mountPoint, flags, mountConfig)
 
-		if err == nil {
+		// This utility is to absorb the error
+		// returned by daemonize.SignalOutcome calls by simply
+		// logging them as error logs.
+		callDaemonizeSignalOutcome := func(err error) {
+			if err2 := daemonize.SignalOutcome(err); err2 != nil {
+				logger.Errorf("Failed to signal to daemon: %v", err2)
+			}
+		}
+
+		markSuccessfulMount := func() {
```

Comment:
Should we merge `markSuccessfulMount` and `markMountFailure` based on `err` parameter?

Reply:
I think it'll just make the caller code harder to read without reducing code much.